lib/archethic/networking/ip_lookup.ex

```diff
@@ -1,25 +1,29 @@
 defmodule Archethic.Networking.IPLookup do
   @moduledoc false
 
-  alias __MODULE__.IPIFY
-  alias __MODULE__.NAT
-
   require Logger
 
+  alias Archethic.Networking
+  alias Archethic.Networking.IPLookup.PublicGateway
+  # alias Archethic.Networking.IPLookup.LocalDiscovery
+
   @doc """
   Get the node public ip with a fallback capability
 
   For example, using the NAT provider, if the UPnP discovery failed, it switches to the IPIFY to get the external public ip
   """
   @spec get_node_ip() :: :inet.ip_address()
-  def get_node_ip do
+  def get_node_ip() do
     provider = get_provider()
 
     ip =
-      case apply(provider, :get_node_ip, []) do
-        {:ok, ip} ->
-          Logger.info("Node IP discovered by #{provider}")
-          ip
+      with {:ok, ip} <- apply(provider, :get_node_ip, []),
+           :ok <- Networking.validate_ip(ip) do
+        Logger.info("Node IP discovered by #{provider}")
+        ip
+      else
+        {:error, reason} when reason == :invalid_ip ->
+          fallback_nat(reason)
```

Comment:
Why are you enforcing a fallback to NAT in that case ?
Each provider should have its way to provide fallback.
So:
- NAT fallback -> IPIFY
- IPIFY -> none
- Static -> none

Reply:
when in dev mode , the provider = get_provider() is Static
```   
with {:ok, ip} <- apply(provider, :get_node_ip, []),
           :ok <- Networking.validate_ip(ip) do
                # With :ok from networking.validate_ip()
                # static is local host which will pass here
                # NAt in docker mode will also pass here
               # IPIFY public not being used in dev mode
        Logger.info("Node IP discovered by #{provider}")
        ip
      else
        {:error, reason} when reason == :invalid_ip ->
          fallback_nat(reason)
 {:error, reason} ->
          fallback(provider, reason)
```

when in prod_mode , WE can have any of the three STATIC,NAT, IPIFIY (in a case)
as per prod.exs
```   
with {:ok, ip} <- apply(provider, :get_node_ip, []),
           :ok <- Networking.validate_ip(ip) do
                # With :ok from networking.validate_ip()
                 # if no errors then IPIFY will pass
                 # nat and static wont pass
        Logger.info("Node IP discovered by #{provider}")
        ip
      else

        {:error, reason} when reason == :invalid_ip ->
            # when in prod Static and nat provider will give invalid ip
            # for both case in prod we, fallback to fallback_nat to go through
           # IPIFY , .
          fallback_nat(reason)
 {:error, reason} ->
    #If error in ipify it will go here
          fallback(provider, reason)
```

Reply:
In this scenario, we wont be able to mock, As we could either have 
```
  {:error, reason} when reason == :invalid_ip ->
          fallback_nat(reason)
```
or provider in config
or use 
```
 defp fallback(LocalDiscovery, reason) do
```
In such case we will need to mock NAT , which will have two behaviour

Reply:
- Which will give error with two behavior's clash.

- Since we cant have params in configs, dev, prod due Regression
We can drop unit tests  for IPlookup.exs

Reply:
marking it as resolved ,
as we have new implementation